src/platform_specific/affinity.unix.cc

```diff
@@ -11,9 +12,12 @@ uint32_t affinity_cores_available() {
 	native_cpu_set affinity_base_mask;
 	assert(pthread_getaffinity_np(pthread_self(), sizeof(cpu_set_t), &affinity_base_mask) == 0 && "Error retrieving base affinity mask.");
 
-	uint32_t count = 0;
+	static uint32_t count = 0;
```

Comment:
That's a bug. This will increase the `count` each time this function is called. To make use of proper static initialization you need to wrap this in a helper function, or maybe a lambda.